File: call_by_reference/call_by_reference.cpp
#include <iostream>
using namespace std;

void swap(int &x, int &y) {
int temp;
temp = x;
x = y;
y = temp;

return;
}

int main () {

int a = 100;
int b = 200;

cout << "Before swap, value of a :" << a << endl;
cout << "Before swap, value of b :" << b << endl;

swap(a, b);

cout << "After swap, value of a :" << a << endl;
cout << "After swap, value of b :" << b << endl;

return 0;
}

File: call_by_reference/perl/call_by_reference.pl
#!/usr/bin/perl -w

my $num1 = 100;
my $num2 = 5;

sub swap{
  $temp = $_[0];
  $_[0] = $_[1];
  $_[1] = $temp;
  return;
}

print "Value of num1 BEFORE swap $num1\n";
print "Value of num2 BEFORE swap $num2\n\n";

swap($num1, $num2);

print "Value of num1 AFTER swap $num1\n";
print "Value of num2 AFTER swap $num2";
